spec.html

```diff
@@ -122,6 +122,29 @@ <h1>
 		</emu-note>
 	</emu-clause>
 
+	<emu-clause id="sec-create-type-error-copy" type="abstract operation">
+		<h1>
+			CreateTypeErrorCopy (
+				_realmRecord_: a Realm Record,
+				_originalError_: an ECMAScript language value
+			): A TypeError object
+		</h1>
+		<emu-alg>
+			1. Let _newError_ be a newly created *TypeError* object.
+			1. NOTE: _newError_ is created in _realmRecord_.
+			1. If Type(_originalError_) is Object, then
+				1. Let _message_ be Completion(Get(_originalError_, "message")).
+				1. If _message_ is not an abrupt completion, then
+					1. If Type(_message_) is String, then
+						1. NOTE: Copying the string message when possible.
+						1. Perform CreateNonEnumerableDataPropertyOrThrow(_newError_, "message", _message_).
+			1. Return _newError_.
+		</emu-alg>
+		<emu-note>
+			For implementations that provide implementation-specific stack for introspection APIs, the TypeError object can be augmented without releaving any information about the stack from within the ShadowRealm, and without providing direct access to any object reference from withtin the ShadowRealm.
```

Comment:
@littledan @ljharb what about now with the note?

Reply:
```suggestion
			For implementations that provide implementation-specific stack for introspection APIs, the TypeError object can be augmented without revealing any information about the stack from within the ShadowRealm, and without providing direct access to any object reference from withtin the ShadowRealm.
```

> without revealing any information about the stack from within the ShadowRealm

i think this will basically shred debuggability. I understand why it's desirable for membranes, but I think that this information should explicitly be *encouraged* to be revealed.

Reply:
I gave this more thoughts and posted a longer explanation in the [original issue](https://github.com/tc39/proposal-shadowrealm/issues/353#issuecomment-1244730856) of what in our opinion the approach should be.